arangod/RocksDBEngine/RocksDBIndexFactory.cpp

```diff
@@ -390,6 +392,8 @@ RocksDBIndexFactory::RocksDBIndexFactory(application_features::ApplicationServer
   emplace("skiplist", skiplistIndexFactory);
   emplace("ttl", ttlIndexFactory);
   emplace("zkd", zkdIndexFactory);
+  emplace(arangodb::iresearch::IRESEARCH_INVERTED_INDEX_TYPE.data(),
```

Comment:
IRESEARCH_INVERTED_INDEX_TYPE is an `std::string_view`, which AFAIK does not guarantee to be null-terminated.

From https://en.cppreference.com/w/cpp/string/basic_string_view/data:

> Unlike std::basic_string::data() and string literals, data() may return a pointer to a buffer that is not null-terminated. Therefore it is typically a mistake to pass data() to a routine that takes just a const CharT* and expects a null-terminated string.

Reply:
IRESEARCH_INVERTED_INDEX_TYPE is initialized as 
`constexpr std::string_view IRESEARCH_INVERTED_INDEX_TYPE = "inverted";`
And string literal is always zero terminated. So here this is safe. But in general yes - string_view could be without zero terminator